### DISCUSSION TITLE
Automatic translation issues : code appears in the translation of our text :(

### DISCUSSION THREAD

**Discussion by amylaboutiqueofficielle** [WeblateOrg/weblate#8628] [Q&A]
Hello, 

We are having some issues using the AUTO TRANSLATE function in Weblate when we translate from French to Spanish. 
Code appears in our Spanish translation when we use the auto translate function ex. [X164X]  which is of course a problem because our consumer will see this when he scrolls our site. 

Does anyone have any advice on how to avoid this? 

Thanks in advance !

- 
[Translation problem Weblate.pdf](https://github.com/WeblateOrg/weblate/files/10444082/Translation.problem.Weblate.pdf)

**Comment by nijel** [comment]
What machine translation service do you use?

It seems to reformat placeables replacements Weblate uses to avoid them being corrupted...

**Reply by amylaboutiqueofficielle** [reply]
Hi ! We use Deepl but we also did a test with Google Translate and the result was the same :(

**Reply by nijel** [reply]
Can you try upgrading Weblate? Since 4.10 we use XML for this in DeepL which should not have any side effects. See #6529

**Reply by nijel** [reply]
For other services, this should be handled better since c1031937dcb46f76150cd21bbfb3db57c1c8dfe6.

**Reply by amylaboutiqueofficielle** [reply]
Thanks ! What do you mean by "this should be handled better since C103193"? We need to change the code in order for it to work?

**Reply by amylaboutiqueofficielle** [reply]
ps. We also have the latest version of Weblate 4.13.1

**Reply by nijel** [reply]
In 4.13.1, DeepL won't produce `[X164X]` tags (it uses XML instead, so it would look like `<x id="164"></x>`). So, either you are not using DeepL or this is a translation from an older version.

Anyway, the latest version is 4.15, you should still consider upgrading.

**Reply by amylaboutiqueofficielle** [reply]
OK i'll check this with the team. Thanks for your quick reply and help on this matter ! Much appreciated :)
